osquery/tables/cloud/azure/azure_instance_tags.cpp

```diff
@@ -0,0 +1,62 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <boost/algorithm/string.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/utils/azure/azure_util.h>
+
+namespace osquery {
+namespace tables {
+namespace pt = boost::property_tree;
+
+QueryData genAzureTags(QueryContext& context) {
+  QueryData results;
+  pt::ptree tree;
+
+  Status s = fetchAzureMetadata(tree);
+
+  if (!s.ok()) {
+    TLOG << "Couldn't fetch metadata: " << s.what();
+  }
+
+  auto tags_str = tree_get(tree, "tags");
+  auto vm_id = tree_get(tree, "vmId");
+  std::vector<std::string> tags;
+
+  boost::split(tags, tags_str, boost::is_any_of(";"));
+
+  for (const auto& tag : tags) {
+    Row r;
+
+    auto colon = tag.find_first_of(':');
```

Comment:
I don't know azure, are tags always key:value?

Reply:
Yes, according to https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service#getting-the-tags-for-the-vm

It looks like a newer version of the metadata service has added `tagsList`, though, which provides a more sensible JSON structure instead of requiring parsing. I'll look into whether we can upgrade to it (i.e., if it's generally available on all regions).